tests/publiccloud/prepare_instance.pm

```diff
@@ -44,6 +44,7 @@ sub run {
     $instance->ssh_opts("");    # Clear $instance->ssh_opts which ombit the known hosts file and strict host checking by default
 
     $instance->network_speed_test();
+    $instance->check_cloudinit() if (cloudinit_enabled);
```

Comment:
so in this job we will run cloud-init 3 times https://openqa.suse.de/tests/15002931 ? I think it is too much

Reply:
The `cloud-init` service is executed every time the machine boots - I didn't change that.

The `cloud-config` file is also loaded every time the machine boots - of course only if it's available - that why this is configurable via `PUBLIC_CLOUD_CLOUD_INIT_FILE`.

Then there are `cloud init scripts` - those are shell scripts and they can be specified to several stages (every boot / only once / every time the instance is (re)created). When you run `cloud-init clean` those stages reset.

So with this pull request our scenario is:
1) Instance is created
2) If the provider supports `cloud-init` we check it.
3) The instance is upgraded
4) If the provider supports `cloud-init` we `clean` it.
5) The instance is rebooted
6) If the provider supports `cloud-init` we check it.

So we **check** how the `cloud-init` went twice: After the instance is created and after upgraded.
And we **clean** the instance once: After the instance is upgraded just before it is rebooted).

Reply:
ok , make sense